### ISSUE TITLE
Improve tower rotation

### ISSUE THREAD

**Issue by sdiaz7** [NestorTejero/ES2016A#161]
### Definition of Done
- Dependency on #162 
- Turrets rotate over time instead of instantly
- Idle rotation when no target is available

**Comment by sdiaz7** [comment]
Try this if you want towers to check if they are facing the target:

```
private bool IsFacingOther(Transform other)
{       
    if (Vector3.Dot(transform.forward, (other.position - transform.position).normalized) == 1)
    {
        return true;    // Return true if facing target.
    }           
    return false;
}
```

**Comment by Nerucius** [comment]
Issue done in parallel with #162.

**Pull Request at #180**

**Comment by Nerucius** [comment]
Merged at #180
